addons/udes_suggest_location/registry/suggest_by_empty_location.py

```diff
@@ -0,0 +1,60 @@
+from .suggest_locations_policy import SuggestLocationPolicy
+from odoo.tools.translate import _
+
+
+class ByEmptyLocation(SuggestLocationPolicy):
+    """Find empty locations of the picking destination location"""
+
+    @classmethod
+    def name(cls):
+        return "by_empty_location"
+
+    def get_values_from_mls(self, mls):
+        """
+        Get the picking destination location from the move lines
+        """
+        return {
+            "location": mls.picking_id.location_dest_id.ensure_one(),
+        }
+
+    def _get_picking_from_dict(self, values):
+        Picking = self.env["stock.picking"]
+        picking_id = values.get("picking_id")
+        if not picking_id:
+            raise ValueError(_("No picking found"))
+        return Picking.browse(picking_id)
+
+    def get_values_from_dict(self, values):
+        """
+        Get the picking destination location from the values
+        """
+        picking = self._get_picking_from_dict(values)
+        return {
+            "location": picking.location_dest_id,
+        }
+
+    def get_locations(self, location, **kwargs):
+        """
+        Search for locations which are children of the picking destination location
+        """
+        Location = self.env["stock.location"]
+        # Add order="id" for performance as we don't care about the order
+
```

Comment:
nit- stray line between comment and code (maybe gap above?)

Reply:
Sorry, not sure what you mean by this, could you explain a bit more please :)

Reply:
Add order id comment refers to the block below the gap. It should be in the same block to indicate what the comment refers to